WordPress/Classes/ViewRelated/Blog/My Site/MySiteOverlaysCoordinator.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+class MySiteOverlaysCoordinator {
+
+    // MARK: - Dependencies
+
+    private var complianceCoordinator: CompliancePopoverCoordinator?
+    private var inAppFeedbackCoordinator: InAppFeedbackPromptPresenting?
```

Comment:
I am not sure what kind of structure you have followed but if VC retains Coordinators (whether directly or via ViewModel), these should be `unowned` variables rather than strongly held. Since we would wish the children Coordinators to be deallocated once their respective VCs are.

Reply:
> these should be unowned variables rather than strongly held. Since we would wish the children Coordinators to be deallocated once their respective VCs are.

Their respective VCs don't hold a reference to these children coordinators, so if we mark them as `unowned` , they will be deallocated immediately.

I have considered instantiating those children coordinators in `MySiteOverlaysCoordinator.presentOverlayIfNeeded`, but I'm planning to write unit tests for `MySiteOverlaysCoordinator` and mock those children coordinators dependencies.

Reply:
I see. So how are the Coordinators retained? A parent coordinator retains a child coordinator and that one retains its children coordinators? If that's the case, do we also need to deallocate them once the screen is dismissed?

So if Coordinator A (CA) is the parent:

CA -> CB -> CC and the ViewController that is linked to CC is dismissed, is there a mechanism to deallocate CC in place?

Reply:
@salimbraksa `MySiteViewController` retains `MySiteOverlaysCoordinator`, which in turns retains those child coordinators.

MySiteVC > MySiteOverlaysCoordinator > Child Coordinators

Also, none of the coordinators retain a View Controller. And the coordinators will be deallocated when MySiteVC is deallocated.

Keep in mind that the coordinators are only responsible for displaying their respective VCs, but they don't retain them.

Reply:
Okay. So if a child coordinator's ViewController is dismissed, the child coordinator itself will not be deallocated because it is being referenced by the parent coordinator.

Because of this problem, I personally prefer that each respective VC retains their Coordinator rather than a parent Coordinator retaining children coordinators. Otherwise coordinators have to replicate a deallocation chain that is provided by Navigation Stack out of the box.

Of course all of this is with the assumption that Coordinators are `class` and not `struct`. Since they often contain states that change over its lifetime, I find class to be a better fit.

Reply:
> I personally prefer that each respective VC retains their Coordinator rather than a parent Coordinator retaining children coordinators.

I think this is not always possible. In cases like the In-App Feedback coordinator, where the VC is just a `UIAlertViewController`, we have to create a subclass to retain the Coordinator.

Additionally, the `InAppFeedbackCoordinator` was inspired by the `CompliancePopoverCoordinator`, and the latter doesn't retain its VC. 

I'm not against your suggestion, but I still don't know what to change exactly. Could you provide a code example to clarify this approach?

Reply:
> If you think in terms of navigation stack, this would be like a whole flow being retained until their root VC is dismissed/deallocated.

I understand, but this is not happening in our case. `MySiteViewController` retains **ONLY** the coordinators of the screens it is showing.

The coordinators are just an alternative to writing the presentation logic within the `MySiteViewController`